site/content/en/references/kustomize/kustomization/buildmetadata/_index.md

```diff
@@ -0,0 +1,308 @@
+---
+title: "buildMetadata"
+linkTitle: "buildMetadata"
+type: docs
+weight: 18
+description: >
+    Specify options for including information about the build in annotations or labels. 
+---
+
+The `buildMetadata` field is a list of strings. The strings can be one of three builtin
+options that add some metadata to each resource about how the resource was built. 
+
+These options are:
+
+- `managedByLabel`
+- `originAnnotations`
+- `transformerAnnotations`
+
+It is possible to set one or all of these options in the kustomization file:
+
+It is possible for to set both `transformerAnnotations` and `originAnnotations`:
+
+```yaml
+buildMetadata: [managedByLabel, originAnnotations, transformerAnnotations]
+```
+
+### Managed By Label
+To mark the resource as having been managed by kustomize, you can specify the `managedByLabel`
+option in the `buildMetadata` field of the kustomization:
+
+```yaml
+buildMetadata: [managedByLabel]
+```
+
+This will add a label `app.kubernetes.io/managed-by` to each resource with the version of kustomize 
+that has built it. For example, given the following input:
+
+kustomization.yaml
+```yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- service.yaml
+buildMetadata: [managedByLabel]
+```
+
+service.yaml
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: myService
+spec:
+  ports:
+  - port: 7002
+```
+
+`kustomize build` will produce a resource with an output like the following:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app.kubernetes.io/managed-by: kustomize-v444.333.222
+  name: myService
+spec:
+  ports:
+  - port: 7002
+```
+
+
+### Origin Annotation
+To retain the information about the origin of a resource, you can specify the `originAnnotations` 
+option in the `buildMetadata` field of the kustomization: 
+
+```yaml
+buildMetadata: [originAnnotations]
+```
+
+When this option is set, generated resources should receive an annotation with key `config.kubernetes.io/origin`,
+containing data about the generator that produced it. If the resource is from the `resources` field, this annotation
+contains data about the file it originated from. 
+
+The possible fields of these annotations are:
```

Comment:
Can we break this out into an example and a separate field list? It's a bit hard to read since the comments aren't syntax highlighted (they're just part of the string).

![image](https://user-images.githubusercontent.com/4789493/152072738-62f3fb9a-7fde-41f7-9e2b-a7fdfb2b3785.png)